experimental/inductor.py

```diff
@@ -0,0 +1,773 @@
+import torch
+from torch import nn, Tensor
+from torch.fx.experimental.proxy_tensor import (
+    make_fx,
+    maybe_disable_fake_tensor_mode,
+)
+from torch._functorch.eager_transforms import functionalize
+from torch._inductor.compile_fx import compile_fx_inner
+from torch._subclasses.fake_tensor import FakeTensorMode, FakeTensor
+from typing import Callable, Dict, Tuple, TypeVar, Union
+from torch._inductor.decomposition import select_decomp_table
+from functools import partial
+from torch.utils._pytree import tree_flatten
+
+
+
+
+inductor_decomps = select_decomp_table()
+
+DIST = True
+DEBUG = True
+
+if DIST:
+    from torch.distributed._tensor.ops.utils import register_prop_rule
+    from torch.distributed._tensor import (
+        Replicate,
+        Shard,
+        DeviceMesh,
+    )
+    from torch.distributed._tensor.dispatch import (
+        operator_dispatch,
+        OpSchema,
+        OutputSharding,
+    )
+    from torch.distributed._tensor.placement_types import DTensorSpec
+    from spmd.compiler.distribute import Schema
+
+    from torch.distributed._tensor import DTensor, ops, distribute_tensor
+    DTensor._op_to_rules['aten.lift_fresh_copy.default'] = ops.tensor_ops.default_prop_rule
+
+    # the rule below may have issues issues:
+    #  - likely doens't cover the case when a shape constant is passed
+    DTensor._op_to_rules['aten.zero.default'] = ops.tensor_ops.prop_create_like
+
+    @register_prop_rule("aten.zeros.default")
+    def prop_aten_zeros_default(op_schema: OpSchema) -> OutputSharding:
```

Comment:
The DTensorMode looks great! Thanks for making this happen! I have one question with regards to those tensor constructors and it's interaction with the DTensorMode:

Most of the nn.Module parameters are initialized with those tensor constructors (i.e. torch.empty), if we implement those rules and use DTensorMode to initialize a model, does it mean that all of the parameters becoming replicated by default?